nbconvert/preprocessors/extractoutput.py

```diff
@@ -80,27 +80,28 @@ def preprocess_cell(self, cell, resources, cell_index):
                 if mime_type in out.data:
                     data = out.data[mime_type]
 
-                    if (
-                        not isinstance(data, text_type)
-                        or mime_type == 'application/json'
-                    ):
+                    if mime_type == 'application/json':
```

Comment:
This is too strict and will break some data types; compare with even a partial listing here: https://jupyter.readthedocs.io/en/latest/reference/mimetype.html

Also this change invalidates the follow-up comment, please keep them in sync.

Reply:
Fair point on the strictness.